src/Components/VersionHistoryPanel.test.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import ShareMock from '../ShareMock'
+import {render} from '@testing-library/react'
+// import {getCommitsForBranch} from '../utils/GitHub'
+// import VersionHistoryPanel from './VersionHistoryPanel'
+
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useNavigate: () => jest.fn(),
+}))
+
+jest.mock('../utils/GitHub', () => ({
+  getCommitsForBranch: jest.fn(),
+}))
+
+describe('VersionsHistoryPanel', () => {
+  it('fetches commits on mount', () => {
+    render(
+        <ShareMock/>,
+        // <ShareMock>
+        //   <VersionHistoryPanel branch="main"/>
+        // </ShareMock>,
+    )
+
+    // await waitFor(() => {
+    //   // expect(getCommitsForBranch).toHaveBeenCalled()
```

Comment:
@oo-bldrs Can you please take a look here. I would like to test if the component is making the call to github, but having trouble with it, since the component is not making the call that is mocked.

Reply:
yes that make sense, i just could not figure out how to have it called and check if it was called.
so resolved with a help of github copilot